internal/kuberesource/parts.go

```diff
@@ -469,8 +469,8 @@ func ServiceMeshProxy() *applycorev1.ContainerApplyConfiguration {
 			WithInitialDelaySeconds(1).
 			WithPeriodSeconds(5).
 			WithFailureThreshold(5).
-			WithTCPSocket(TCPSocketAction().
-				WithPort(intstr.FromInt(15006))),
+			WithExec(ExecAction().
+				WithCommand("test", "-f", "/ready")),
```

Comment:
I'm fine with having the file-based solution, but is there any reasoning behind using this and not e.g. a `wait -S SIGCONT`? PID propagation to the Go process?

Reply:
What I'm trying to do here is to tell k8s that my container is ready. There's only a limited amount of ways to do so, and using the network probes does not work (in fact, is the bug I'm trying to solve). This leaves me with the exec probe.

I don't really understand what you're suggesting:

1. What is `wait -S`?
2. Who should call it?

Reply:
I understand what you're doing with the exec probe, but I think that we're misusing the existence of the `/ready` file as a "poor man's IPC" here, while "real" (signal-based) IPC would come to my mind much more naturally here. I'm fine with the file-based solution, as the risk of races etc. would seem very low in this case. I just wondered about the file-based solution here, as it seemed a little unconventional to me.

To answer your questions:

1. `wait(1p)` waits for the specified signal
2. The execProbe, which would halt until the signal is received and then exit successfully

Reply:
1. Do you have a reference for this usage of wait? I can't find it in `man 1p wait`, nor in the POSIX spec.
2. I don't think probes are meant to be blocking.

Reply:
Sorry for the confusion here. I got `trap` and `wait` scrambled around, but I swear there was some thing to wait for an explicit signal. Let's just go with the file-based approach here